homework_25/src/main/java/org/jdbc/operations/impl/LessonDao.java

```diff
@@ -0,0 +1,135 @@
+package org.jdbc.operations.impl;
+
+import org.jdbc.entity.Homework;
+import org.jdbc.entity.Lesson;
+import org.jdbc.exceptions.JdbcOperationException;
+import org.jdbc.operations.HomeworkDao;
+import org.jdbc.operations.LessonDaoOperations;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class LessonDao implements LessonDaoOperations<Lesson, Long> {
+    private DataSource dataSource;
+
+    public LessonDao(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public Lesson insertLesson(final Lesson lesson) {
+        Objects.requireNonNull(lesson);
+        if (lesson.getId() != null) {
+            throw new JdbcOperationException("ID mustn`t be provided during the insert operation!");
+        }
+
+        var sql = """
+                INSERT INTO lesson (name, homework_id) VALUES (?, ?)
+                """;
+
+        try (Connection connection = this.dataSource.getConnection();
+             PreparedStatement prepStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            int idx = 1;
+            prepStatement.setString(idx++, lesson.getName());
+            prepStatement.setLong(idx, lesson.getHomework().getId());
+
+            int rowsInserted = prepStatement.executeUpdate();
+            if (rowsInserted < 1) {
+                throw new JdbcOperationException("No rows were added into database, input lesson = %s".formatted(lesson));
+            }
+            ResultSet generatedKeys = prepStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getLong(1);
```

Comment:
`1` looks like a magic number. In this particular case please create a variable `generatedKeysIndex` or fetch the key by its string representation.